packages/libs/client-utils/src/nodejs/yaml-converter.ts

```diff
@@ -85,91 +87,93 @@ export const getPartsAndHolesInCtx = (
 };
 
 // Responsible for replacing holes with values
-export const replaceHoles = (
-  partsAndHoles: PartsAndHoles,
-  args: Record<string, string | number>,
-) => {
-  const [parts, holes] = partsAndHoles;
-  const allParts = zip(parts, holes);
-  return allParts
-    .map((partOrHole, index) => {
-      if (typeof partOrHole === 'string') {
-        // it's a part
-        return partOrHole;
-      } else {
-        // Currently we are unaware of the difference between {{}} and {{{}}} so we treat them the same: as a literal hole
-        if ('literal' in partOrHole) {
-          // it's a literal hole
-          if (!(partOrHole.literal in args)) {
-            throw new Error(
-              `argument to fill hole for ${partOrHole.literal} is missing in ${
-                allParts[index - 1]
-              }{{${partOrHole.literal}}}${allParts[index + 1]}}`,
-            );
+export const replaceHoles =
+  (args: Record<string, string | number>) => (partsAndHoles: PartsAndHoles) => {
+    const [parts, holes] = partsAndHoles;
+    const allParts = zip(parts, holes);
+    return allParts
+      .map((partOrHole, index) => {
+        if (typeof partOrHole === 'string') {
+          // it's a part
+          return partOrHole;
+        } else {
+          // Currently we are unaware of the difference between {{}} and {{{}}} so we treat them the same: as a literal hole
+          if ('literal' in partOrHole) {
+            // it's a literal hole
+            if (!(partOrHole.literal in args)) {
+              throw new Error(
+                `argument to fill hole for ${
+                  partOrHole.literal
+                } is missing in ${allParts[index - 1]}{{${
+                  partOrHole.literal
+                }}}${allParts[index + 1]}}`,
+              );
+            }
+
+            return args[partOrHole.literal];
           }
-
-          return args[partOrHole.literal];
         }
-      }
-    })
-    .join('');
+      })
+      .join('');
+  };
+
+const loadYaml = (filledYamlString: string) => {
+  return yaml.load(filledYamlString) as ITemplateTransaction;
 };
 
 // Responsible for replacing holes in a context with values
 export const replaceHolesInCtx = (args: Record<string, string | number>) => {
```

Comment:
@alber70g @nil-amrutlal-dept 
Do we really need this function? Why not just using `replaceHoles` directly?
also in the `createPactCommandFromTemplate` you called this function but actually only the filledYamlString is used.

Reply:
I will have a look at this when I have some time. Added an issue on the board.
Need to recheck the flow to see if I need it

Reply:
I think it mostly has to do with the functional pipe, the next method passed to is `parseYamlToKdaTx` and it expects the entire ctx object, for the new method I didn't need the entire ctx since it won't try to load any files.